idaes_examples/notebooks/docs/unit_models/custom_unit_models/aqueous_property.py

```diff
@@ -0,0 +1,303 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES).
+#
+# Copyright (c) 2018-2023 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory,
+# National Technology & Engineering Solutions of Sandia, LLC, Carnegie Mellon
+# University, West Virginia University Research Corporation, et al.
+# All rights reserved.  Please see the files COPYRIGHT.md and LICENSE.md
+# for full copyright and license information.
+#################################################################################
+
+# Import Python libraries
+import logging
+
+import idaes.logger as idaeslog
+from idaes.core.util.initialization import fix_state_vars, revert_state_vars
+
+# Import Pyomo libraries
+from pyomo.environ import (
+    Param,
+    Set,
+    Var,
+    NonNegativeReals,
+    units,
+    Expression,
+    PositiveReals,
+)
+
+# Import IDAES cores
+from idaes.core import (
+    declare_process_block_class,
+    MaterialFlowBasis,
+    PhysicalParameterBlock,
+    StateBlockData,
+    StateBlock,
+    MaterialBalanceType,
+    EnergyBalanceType,
+    Solute,
+    Solvent,
+    LiquidPhase,
+)
+from idaes.core.util.model_statistics import degrees_of_freedom
+
+# Some more information about this module
+__author__ = "Javal Vyas"
+
+# Set up logger
+_log = logging.getLogger(__name__)
+
+
+@declare_process_block_class("AqPhase")
+class PhysicalParameterData(PhysicalParameterBlock):
+    """
+    Property Parameter Block Class
+
+    Contains parameters and indexing sets associated with properties for
+    liquid Phase
+
+    """
+
+    def build(self):
+        """
+        Callable method for Block construction.
+        """
+        super().build()
+
+        self._state_block_class = AqPhaseStateBlock
+
+        # List of valid phases in property package
+        self.Aq = LiquidPhase()
+
+        # Component list - a list of component identifiers
+        self.NaCl = Solute()
+        self.KNO3 = Solute()
+        self.CaSO4 = Solute()
+        self.H2O = Solvent()
+
+        self.solutes = Set(initialize=["NaCl", "KNO3", "CaSO4"])
```

Comment:
You should not need to create this is you are using `Solute` components above. Also, do you need this later at all, as not all property packages will have this.

Reply:
I do use this in the state block, as the indexed set, for `self.conc_mass_comp` variable. I believe I can change this to `dissolved_solutes` and that should help this case.

Reply:
My first point is that you don't need to make this yourself - it should already exist for you. See line 370-375: https://github.com/IDAES/idaes-pse/blob/main/idaes/core/base/components.py
My second point is that you should be careful how you use this (mostly, don't try to use this outside the property package), as it is not something that all property packages will have.